server/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -504,10 +511,14 @@ public static <T> void buildReference(RelationName relationName,
         ref = columnDefinition.isGenerated()
             ? new GeneratedReference(ref, columnDefinition.formattedGeneratedExpression(), null)
             : ref;
-        references.add(ref);
+
+        references.putIfAbsent(ref.column().fqn(), ref);
```

Comment:
Could use the `ColumnIdent` as key instead of `fqn` as it's less ambiguous. `ColumnIdent` clearly communicates the format, `String` doesn't show its format (whether it is in dotted notation, or the sql-subscript notation, or if it can only be top-level columns)

Reply:
done in https://github.com/crate/crate/pull/13383/commits/41217de4ff7ce02f31db5f5fa786b7e23bd5fa18